proposals/NNNN-align-VidepStreamingParameter-with-capability.md

```diff
@@ -0,0 +1,124 @@
+
+The issue is realized when implementing [SDL-0274](0274-add-preferred-FPS.md) for sdl_java_suite. In [SDL-0274](0274-add-preferred-FPS.md), "iOS mobile proxy changes" section states as follows:
+
+>Later part in the same method (didEnterStateVideoStreamStarting), current videoEncoderSettings are overwritten by customEncoderSettings.
+>
+>However, for frame rate, we should take lower value approach if frame rate in customEncoderSettings is higher than preferred FPS
+
+We should take the same approach, i.e. taking lower preferred FPS value for Java Suite library too, but during the code review, we found that the same approach should be taken for some other parameters as well.
+
+This proposal addresses how developer's settings should be aligned with `VideoStreamingCapability`.
+
+## Proposed solution
+
+As mentioned in Motivation section, we should take lower value approach for relevant video streaming settings if the value specified by developer is higher than capabilities' value.
+
+### Detailed design for iOS library
+
+iOS library takes `customVideoEncoderSettings`, but values relevant to VideoStreamingCapability are `AverageBitRate` and `ExpectedFrameRate`.
+
+Those values should be aligned as follows:
+
+SDLVideoStreamingCapability.m:
+
+```objc
+- (void)didEnterStateVideoStreamStarting {
+		...
+        // Apply customEncoderSettings here. Note that value from HMI (such as maxBitrate) will be overwritten by custom settings
+        // (Exception: ExpectedFrameRate, AverageBitRate)
+        for (id key in self.customEncoderSettings.keyEnumerator) {
+            // do NOT override framerate or average bitreate if custom setting is higher than current setting.
+            if ([(NSString *)key isEqualToString:(__bridge NSString *)kVTCompressionPropertyKey_ExpectedFrameRate] ||
+                [(NSString *)key isEqualToString:(__bridge NSString *)kVTCompressionPropertyKey_AverageBitRate]) {
+                if ([self.customEncoderSettings valueForKey:key] < self.videoEncoderSettings[key]) {
+                    self.videoEncoderSettings[key] = [self.customEncoderSettings valueForKey:key];
+                }
+            } else {
+                self.videoEncoderSettings[key] = [self.customEncoderSettings valueForKey:key];
+            }
+        }
+		...
+}
+```
+
+### Detailed design for Java Suite library
+
+Java Suite library updates custom `VideoStreamingParameters` when `VideoStreamingCapability` is given by head unit. `VideoStreamingParameter#update` method does the job.
+
+For Resolution and Scale, the `update()` method already takes care of capability as well as making some adjustment when `vehicleMake` is specified, like below:
+
+VideoStreamingParameter.java (**no change seems to be needed for Resolution and Scale**)
+
+```java
+public void update(VideoStreamingCapability capability, String vehicleMake) {
+	...
+    // For resolution and scale, the capability values should be taken than parameters specified by developers.
```

Comment:
Sorry, that comment was somewhat broken. I changed the comment to:
// For resolution and scale, the capability values should be taken, rather than taking parameters specified by developers.